src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -170,30 +170,32 @@ private static byte[] TryGetByteArrayFromStream(Stream data)
         }
 
         /// <summary>
-        /// If the MemoryStream was created with publiclyVisible=true, then we can access its buffer
+        /// If the MemoryStream was created with publiclyVisible=true, then we can try to access its buffer
         /// directly and save allocations in StreamReader. The input MemoryStream is not closed on exit.
         /// </summary>
         /// <returns>True if a byte array could be created.</returns>
         private static bool TryGetByteArrayFromMemoryStream(MemoryStream data, out byte[] buffer)
         {
             Debug.Assert(data.Position == 0);
 
-            try
+            if (PortableShim.MemoryStream.TryGetBuffer != null)
             {
-                if (PortableShim.MemoryStream.GetBuffer != null)
```

Comment:
https://msdn.microsoft.com/en-us/library/system.io.memorystream.trygetbuffer(v=vs.110).aspx

Added in .NET 4.6.